### PR TITLE
📝 Docs: Files Source setup steps update

### PR DESCRIPTION
**Pull request by ChristoGrab**
## What
Updated the setup steps in the Files source documentation.

## How
- Added standard connector setup steps.
- Re-arranged document flow to follow the order of input fields in Airbyte UI.
- Expanded section on provider-specific fields by noting required/optional nature of all fields, adding some context and adding more links to external sources where helpful.
- Merged duplicate steps for SSH, SCP, and SFTP into a single header.
- Consolidated and added `env:local` tag to all setup instructions for using local filesystem.
- Consolidated provider-specific info to remove duplicate statements and shift important information to the relevant sections in setup steps.
- Added JSONL to table of supported formats.
- General clarification/typo fixes.

### CODE REVIEW

File: docs/integrations/sources/file.md [hunk 1]
```diff
@@ -5,71 +5,99 @@ This page contains the setup guide and reference information for the Files sourc
 ## Prerequisites
 
 - URL to access the file
-- Format
-- Reader options
-- Storage Providers
+- File Format
+- Storage Provider
+- Reader Options (Optional)
```
Comment (sherifnada) [inline docs/integrations/sources/file.md:10]: I think of pre-requisites as "things you need to go get before starting to setup this connector". In that sense, the only requirement is "A file hosted on SFTP, S3, GCS, HTTPS, or SFTP server"

File: docs/integrations/sources/file.md [hunk 2]
```diff
@@ -5,71 +5,99 @@ This page contains the setup guide and reference information for the Files sourc
...
+  
+Enter the _username_ associated with your account on the remote server.
+- `Password` (Optional) 
+  
+**If required by the remote server**, enter the _password_ associated with your user account. Otherwise, leave this field blank.
+- `Port` (Optional) 
+
+Specify the _port number_ to use for the connection. The default port is usually 22. However, if your remote server uses a non-standard port, you can enter the appropriate port number here.
+
+<!-- env:local -->
```
Comment (sherifnada) [inline docs/integrations/sources/file.md:74]: I'm not sure `env:local` is a thing -- did you see it somewhere else? the only two i know of are oss and cloud

File: docs/integrations/sources/file.md [hunk 3]
```diff
@@ -5,71 +5,99 @@ This page contains the setup guide and reference information for the Files sourc
...
+#### Local Filesystem (Airbyte Open Source only)
+- `Storage`
+
+:::caution 
+Currently, the local storage URL for reading must start with the local mount "/local/" until we implement more advanced docker mounting options.
```
Comment (sherifnada) [inline docs/integrations/sources/file.md:79]: There are currently no plans to expand the local mode
```suggestion
Currently, the local storage URL for reading must start with the local mount "/local/".
```

File: docs/integrations/sources/file.md (hunk 2 above)
Comment (sherifnada) [reply docs/integrations/sources/file.md:74]: This should probably just be oss since cloud doesn't support local

**Review by sherifnada — APPROVED** [review]
looks great! Much more accessible. Just a few comments but you're gtg afterwards

File: docs/integrations/sources/file.md (hunk 2 above)
Comment (ChristoGrab) [reply docs/integrations/sources/file.md]: Whoops my bad! I just got mixed up with the env name for oss 🤦‍♂️

File: docs/integrations/sources/file.md (hunk 3 above)
Comment (ChristoGrab) [reply docs/integrations/sources/file.md]: Done!

File: docs/integrations/sources/file.md (hunk 1 above)
Comment (ChristoGrab) [reply docs/integrations/sources/file.md]: Gotcha, fixed it 👍

**Pull request merged**